examples/conanfile/package_id/settings_and_options.rst

```diff
@@ -0,0 +1,242 @@
+        version = "1.0.0"
+        # No settings this time
+
+Or, if the tool is platform specific we can just keep the OS and architecture information:
+
+.. code-block:: python
+
+    from conan import ConanFile
+
+    class MyScripts(ConanFile):
+        name = "my-scripts"
+        version = "1.0.0"
+        settings = "os", "arch"
+
+How options influence the package ID
+------------------------------------
+
+Options are used to specify characteristics that are particular to a single recipe, contrasting with settings that generally remain consistent across recipes within a project.
+They are usually a set of particular characteristics of a library executable or conan package may have.
+
+For example, a `shared` option is a very common option used in recipes that can produce shared libraries. However, it could not be a setting as not all recipes produce shared libraries.
+
+.. code-block:: python
+
+    from conan import ConanFile
+
+    class Pkg(ConanFile):
+        name = "pkg"
+        version = "1.0.0"
+        options = {"shared": [True, False]}
+        default_options = {"shared": True}
+
+As in the previous case with settings, the different values of an option will influence the package ID and therefore, generate different packages depending on it.
+
+.. code-block:: bash
+   :emphasize-lines: 3, 7
+
+    $ conan create . --options shared=True
+    ...
+    pkg/1.0.0: Package '1744785cb24e3bdca70e27041dc5abd20476f947' created
+
+    $ conan create . --options shared=False
+    ...
+    pkg/1.0.0: Package '55c609fe8808aa5308134cb5989d23d3caffccf2' created
+
+In the same way, there might be "options" that are needed as input in a recipe to generate a package which shouldn't be taken into account in the package ID.
+An example of this could be an option to control something that during the build phase but that does not influence the package result, like the *verbosity* of a compilation.
+In that case, the recipe should remove the option in the :ref:`package_id() method <reference_conanfile_methods_package_id>`:
+
+However, the general advice is that **options should always affect the package ID**, and in case we would like to have an input to the recipe that should **not** affect it,
+it should be done via the :ref:`conf section <reference_config_files_profiles_conf>` of your profile. Then in the recipe we should just add:
+
+.. code-block:: python
+
+    from conan import ConanFile
+
+    class MyPkg(ConanFile):
+        name = "my-pkg"
+        version = "1.0.0"
+
+        def build(self):
+            verbosity = self.conf.get("user.my-pkg:verbosity")
+            self.output.info(f"Using verbosity level: {verbosity})
+            ...
+
+.. code-block:: text
+    :caption: *myprofile*
+
+    [conf]
+    user.my-pkg:verbosity=silent
+
+That way the package ID will be not affected, the recipe will be cleaner (without irrelevant options for package ID) and the input is easily managed via the profile's conf section.
```

Comment:
I think it would be good to add a read more section also here, linking back to other places of interest, like this one: https://docs.conan.io/2/tutorial/creating_packages/configure_options_settings.html

Reply:
Also this one: https://docs.conan.io/2/reference/binary_model/package_id.html and linking back would be good too

Reply:
done